python/tests/io/test_pyarrow.py

```diff
@@ -264,3 +265,27 @@ def test_deleting_s3_file_not_found():
             PyArrowFileIO().delete("s3://foo/bar.txt")
 
         assert "Cannot delete file, does not exist:" in str(exc_info.value)
+
+
+def test_convert_iceberg_schema_to_pyarrow_schema(table_schema_nested: Schema):
```

Comment:
Needs a bit more testing.

Reply:
Added individual tests for all of the types

Reply:
I would love to also add more roundtrip tests for this 👍🏻 But would be nice to do this as soon as we have write support, so we can write a table, and read it afterward to check if everything works